### PR TITLE
Add analyzer to replace Assert.Collection with Assert.Empty when no Action is specified

### PR DESCRIPTION
**Pull request by ErikSchierboom**
Refs https://github.com/xunit/xunit/issues/1256

### CODE REVIEW

File: src/xunit.analyzers/Descriptors.cs [hunk 1]
```diff
@@ -198,5 +198,10 @@ internal static class Descriptors
             "Do not use boolean check to check for string equality",
             "Do not use {0} to check for string equality.",
             Categories.Assertions, DiagnosticSeverity.Warning, isEnabledByDefault: true);
+
+        internal static DiagnosticDescriptor X2011_AssertCollectionEmptyCheckShouldNotUseCollectionCheckFixer { get; } = new DiagnosticDescriptor("xUnit2011",
```
Comment (marcind) [inline src/xunit.analyzers/Descriptors.cs]: Accidental "Fixer" suffix?

File: src/xunit.analyzers/AssertCollectionEmptyCheckShouldNotUseCollectionCheckFixer.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        public sealed override ImmutableArray<string> FixableDiagnosticIds { get; } = ImmutableArray.Create(Descriptors.X2011_AssertCollectionEmptyCheckShouldNotUseCollectionCheckFixer.Id);
+
+        public sealed override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var invocation = root.FindNode(context.Span).FirstAncestorOrSelf<InvocationExpressionSyntax>();
+            
+            context.RegisterCodeFix(
```
Comment (marcind) [inline src/xunit.analyzers/AssertCollectionEmptyCheckShouldNotUseCollectionCheckFixer.cs]: I think there should be an alternative fixer for this, which would add an element inspector. If you don't have the bandwidth/interest to implement that we could open a follow up issue instead.

**Review by marcind — CHANGES_REQUESTED** [review]

File: src/xunit.analyzers/Descriptors.cs (hunk 1 above)
Comment (ErikSchierboom) [reply src/xunit.analyzers/Descriptors.cs]: Whoops. I'll fix it

File: src/xunit.analyzers/AssertCollectionEmptyCheckShouldNotUseCollectionCheckFixer.cs (hunk 2 above)
Comment (ErikSchierboom) [reply src/xunit.analyzers/AssertCollectionEmptyCheckShouldNotUseCollectionCheckFixer.cs]: Excellent suggestion. I'll add it to the PR

Comment (ErikSchierboom) [reply src/xunit.analyzers/AssertCollectionEmptyCheckShouldNotUseCollectionCheckFixer.cs]: @marcind I've added a second code fix to add an element inspector. At the moment, the inspector is just an empty lamdba: `x => { }`. Is that okay or would you like to use something else?

I've also slightly renamed the code fix to not have it tied to the empty collection check.

**Review by marcind — APPROVED** [review]

**Pull request merged**